File: 0002-add-two-numbers/0002-add-two-numbers.c
/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     struct ListNode *next;
 * };
 */
struct ListNode* addTwoNumbers(struct ListNode* l1, struct ListNode* l2){
    int c=0,s,i;
    struct ListNode *r=NULL,*t;
    for(int i=1; l1!=NULL || l2!=NULL ;)
    {   
        s = c + (l1?l1->val:0)+(l2?l2->val:0);
        c = s/10, s%=10; 
        // printf("%d  ",s);

    // if(s==0)
    // {
    //     l1 = (struct ListNode*)malloc(sizeof(struct ListNode));
    //     l1->next = NULL;
    //     l1->val=s;
    //     return l1;
    // }
    // else
    // {
        if(r==NULL)
        {
        r = (struct ListNode*)malloc(sizeof(struct ListNode));
        r->next = NULL;
        r->val = s;
        t=r;
        // printf("%d",r->val);
        }
        else
        {
        t->next = (struct ListNode*)malloc(sizeof(struct ListNode));
        t->next->next = NULL;
        t->next->val = s;
        t=t->next;
        // printf("%d ",t->val);
        }
    // }
    if(l1)
        l1=l1->next;
    if(l2)
        l2=l2->next;

    }
    if(c>0)
    {
        t->next = (struct ListNode*)malloc(sizeof(struct ListNode));
        t->next->next = NULL;
        t->next->val = c;
    }
 
    return r;
}

File: 0002-add-two-numbers/0002-add-two-numbers.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
 import java.math.BigInteger;
class Solution {
    public ListNode addTwoNumbers(ListNode l1, ListNode l2) {
       BigInteger sum = BigInteger.ZERO, n1 = BigInteger.ZERO, n2 = BigInteger.ZERO, mul = BigInteger.ONE;

        while (l1 != null) {
            n1 = n1.add(mul.multiply(BigInteger.valueOf(l1.val)));
            mul = mul.multiply( BigInteger.valueOf(10));
            l1 = l1.next;
        }
        mul = BigInteger.ONE;
        while (l2 != null) {
            n2 = n2.add(mul.multiply(BigInteger.valueOf(l2.val)));
            mul = mul.multiply( BigInteger.valueOf(10));
            l2 = l2.next;
        }
        sum = n2.add(n1);
        System.out.println(sum);
        l1 = new ListNode(Integer.parseInt(sum.mod(BigInteger.TEN).toString()),null);
        l2 = l1;
        sum= sum.divide(BigInteger.TEN);
        while (!sum.equals(BigInteger.ZERO)){
            l1.next = new ListNode();
            l1 = l1.next;
            l1.val = Integer.parseInt(sum.mod(BigInteger.TEN).toString());
            sum=sum.divide(BigInteger.TEN);
        }
        return l2;
    }
}